scapy/compat.py

```diff
@@ -48,6 +48,18 @@ def cmp(a, b):
     """Old Python 2 function"""
     return (a > b) - (a < b)
 
+def lambda_tuple_converter(func):
+    """
+    Converts a Python 2 function as
+      lambda (x,y): x + y
+    In the Python 3 format:
+      lambda x,y : x + y
+    """
+    if func is None:
+        return func
+    if func.__code__.co_argcount == 1: # Detected format lambda (x,y): ... or lambda x: x[0]...
+        return lambda *args: func(args[0] if len(args) == 1 else args)
+    return func
 
```

Comment:
if func is not None and  func.__code__.co_argcount == 1:
    return lambda *args: func(args[0] if len(args) == 1 else args)
else:
    return func